docs/pages/machine-id/deployment/spacelift.mdx

```diff
@@ -0,0 +1,286 @@
+example-bot Bot
+```
+
+## Step 2/3. Create a role and Machine ID bot
+
+Create `example-bot-role.yaml`, which declares a Teleport role that we will
+assign to the bot user for Spacelift. `tbot` generates short-lived credentials
+that grant the user access to this role, allowing Spacelift to manage dynamic
+Teleport resources using Terraform:
+
+```yaml
+kind: role
+version: v5
+metadata:
+  name: example-bot
+spec:
+  allow:
+    rules:
+    - resources:
+      - app
+      - cluster_auth_preference
+      - cluster_networking_config
+      - db
+      - device
+      - github
+      - login_rule
+      - oidc
+      - okta_import_rule
+      - role
+      - saml
+      - session_recording_config
+      - token
+      - trusted_cluster
+      - user
+      verbs:
+      - create
+      - read
+      - update
+      - delete
+      - list
+  deny: {}
+  options: {}
+```
+
+This role grants access to create, update, delete, and list a number of Teleport
+resources. You may wish to remove resources that you do not intend to configure
+with Terraform from this list to reduce blast radius. See the [Teleport Role
+Reference](../../access-controls/reference.mdx#rbac-for-dynamic-teleport-resources)
+for the dynamic resources you can grant access to in a Teleport role.
+
+Create this role by applying the manifest:
+
+```
+$ tctl create -f example-bot-role.yaml
+```
+
+Create the bot, specifying the role and token that you have created:
+
+```code
+$ tctl bots add example --roles=example-bot --token=example-bot
+```
+
+## Step 3/3. Configure your Spacelift stack
+
+While following this step, you will modify your git repo to:
+- Configure Spacelift to authenticate the Teleport Terraform provider as a bot
+  user using credentials generated by Machine ID.
+- Create dynamic Teleport resources using your git repo.
+
+Before continuing, clone your GitHub repository. In the clone, check out a
+branch from your main branch. 
+
+### Configure Spacelift to authenticate as a bot user
+
+Now that the bot has been successfully created, you now need to configure your
+Spacelift stack to authenticate as this bot using `tbot` and then configure
+the Terraform provider to use the credentials produced by `tbot`.
+
+To help with this, Teleport publishes a custom Spacelift container image. This
+image is based on the default image provided by Spacelift but additionally
+includes `tbot`.
+
+Within the repository linked to your Spacelift stack, create
+`.spacelift/config.yaml` to specify the `teleport-spacelift-runner` image and to
+include a `before_init` step that invokes `tbot` to produce credentials:
+
+```yaml
+version: "1"
+stack_defaults:
+  runner_image: public.ecr.aws/gravitational/teleport-spacelift-runner:(=teleport.version=)
+  before_init:
+  - |
+    tbot start --oneshot \
+      --data-dir=memory:// \
+      --auth-server teleport.example.com:443 \
+      --join-method spacelift --token example-bot \
```

Comment:
nit: put `--token` on its own line so we have one flag per line. On first read I missed this as I quickly assumed this whole line was for the `--join-method`.

Reply:
[7d242e4](https://github.com/gravitational/teleport/pull/34935/commits/7d242e4ce8d207ec63c853c485be87f2635b9274)